README_comments.md

```diff
@@ -0,0 +1,247 @@
+```
+
+
+### As root user on Linux
+
+If you have a Linux machine, as well as root privileges, (e.g., if you plan to
+run the workflow on your own computer), you can execute the following command
+to include Singularity in the Conda environment:
+
+```bash
+conda env create -f install/environment.root.yml
+```
+
+## 4. Activate environment
+
+Activate the Conda environment with:
+
+```bash
+conda activate zarp
+```
+
+# Extra installation steps (optional)
+
+## 5. Non-essential dependencies installation
+
+Most tests have additional dependencies. If you are planning to run tests, you
+will need to install these by executing the following command _in your active
+Conda environment_:
+
+```bash
+conda env update -f install/environment.dev.yml
+```
+
+## 6. Successful installation tests
+
+We have prepared several tests to check the integrity of the workflow and its
+components. These can be found in subdirectories of the `tests/` directory. 
+The most critical of these tests enable you to execute the entire workflow on a 
+set of small example input files. Note that for this and other tests to complete
+successfully, [additional dependencies](#installing-non-essential-dependencies) 
+need to be installed. 
+Execute one of the following commands to run the test workflow 
+on your local machine:
+* Test workflow on local machine with **Singularity**:
+```bash
+bash tests/test_integration_workflow/test.local.sh
+```
+* Test workflow on local machine with **Conda**:
+```bash
+bash tests/test_integration_workflow_with_conda/test.local.sh
+```
+Execute one of the following commands to run the test workflow 
+on a [Slurm][slurm]-managed high-performance computing (HPC) cluster:
+
+* Test workflow with **Singularity**:
+
+```bash
+bash tests/test_integration_workflow/test.slurm.sh
+```
+* Test workflow with **Conda**:
+
+```bash
+bash tests/test_integration_workflow_with_conda/test.slurm.sh
+```
+
+> **NOTE:** Depending on the configuration of your Slurm installation you may
+> need to adapt file `slurm-config.json` (located directly under `profiles`
+> directory) and the arguments to options `--cores` and `--jobs`
+> in the file `config.yaml` of a respective profile.
+> Consult the manual of your workload manager as well as the section of the
+> Snakemake manual dealing with [profiles].
+
+# Running the workflow on your own samples
+
+1. Assuming that your current directory is the repository's root directory,
+create a directory for your workflow run and move into it with:
+
+    ```bash
+    mkdir config/my_run
+    cd config/my_run
+    ```
+
+2. Create an empty sample table and a workflow configuration file:
+
+    ```bash
+    touch samples.tsv
+    touch config.yaml
+    ```
+
+3. Use your editor of choice to populate these files with appropriate
+values. Have a look at the examples in the `tests/` directory to see what the
+files should look like, specifically:
+
+    - [samples.tsv](tests/input_files/samples.tsv)
```

Comment:
The sample entries in this file imply a certain directory structure. If I were to use these files as templates, the directory structure for my_run doesn't match and this makes it confusing. For example, it is necessary for the input files to have a path ../input_files/* relative to the current directory?  And if the current directory was set up as specified above, namely config/my_run, then the path to the sample files in the input_files directory would be ../../input_files/*, so not as in the test example. Would be good to revisit the directory structure and make sure that it's logical and consistent.